File: src/store.rs
use std::sync::{Arc, RwLock};

use crate::models::Post;

pub type PostStore = Arc<RwLock<Vec<Post>>>;

pub fn new_store() -> PostStore {
    let posts = vec![
        Post {
            id: "pst-9m4e2mr0ui3e8a215n4g".to_string(),
            title: "Hello, DevOps!".to_string(),
        },
        Post {
            id: "pst-9m4e2mr0ui3e8a215n4f".to_string(),
            title: "CI/CD with GitHub Actions".to_string(),
        },
    ];
    Arc::new(RwLock::new(posts))
}

File: src/store.ts
import type { Post } from './models.js';

export class PostStore {
  private posts: Post[];

  constructor(initial: Post[] = []) {
    this.posts = [...initial];
  }

  list(): Post[] {
    return [...this.posts];
  }

  get(id: string): Post | undefined {
    return this.posts.find((p) => p.id === id);
  }

  create(post: Post): void {
    this.posts.push(post);
  }
}

export function createSeededStore(): PostStore {
  return new PostStore([
    { id: 'pst-seed-1', title: 'Hello, DevOps!' },
    { id: 'pst-seed-2', title: 'CI/CD with GitHub Actions' },
  ]);
}
